Common/Product/SharedProject/HierarchyNode.cs

```diff
@@ -590,7 +590,7 @@ internal abstract class HierarchyNode :
                     break;
 
                 case __VSHPROPID.VSHPROPID_Name:
-                    result = this.Caption;
+                    result = CommonUtils.GetFileOrDirectoryName(this.Url);

```

Comment:
If I look though this file for other uses for `this.caption`, then I see it used for `SaveName is the name shown in the Save and the Save Changes dialog boxes` and `GetEditLabel: Called by the shell to get the node caption when the user tries to rename from the GUI`, and `GetRelationalName`. Seems like all of these probably should be using the proper filename/url also, and not the labeled caption.